extensions/amp-form/0.1/amp-form.js

```diff
@@ -216,6 +221,29 @@ export class AmpForm {
   }
 
   /**
+   * Returns form data as an object.
+   * @return {!Object}
+   * @private
+   */
+  getFormAsObject_() {
```

Comment:
Thanks it looks a lot similar to what we do, few differences are they filter out some of the unnecessary input types. And they seem to process the input value to replace `newline` and `carriage return` characters with `\r\n`. Not sure if we want to do this.

```
  rsubmittable.test( this.nodeName ) && !rsubmitterTypes.test( type ) &&
val.replace( rCRLF, "\r\n" )
```

```javascript
	rsubmitterTypes = /^(?:submit|button|image|reset|file)$/i,
	rsubmittable = /^(?:input|select|textarea|keygen)/i;
```

They also seem to be filtering out `submit` type which I don't think is accurate in term of what the browser default does, the browser will actually serialize and append to params the name and value of the submit button the user clicked.

I can add some logic to filter out the other types.